src/components/pages/services/store/actionCreators.tsx

```diff
@@ -0,0 +1,35 @@
+import { ITask, TaskAction, DispatchType } from "../types/type.d";
+import * as actionTypes from "./actionTypes";
+
+export function addTask(task: ITask) {
+  const action: TaskAction = {
+    type: actionTypes.ADD_TASK,
+    task,
+  };
+
+  return simulateHttpRequest(action);
+}
+
+export function updateTask(task: ITask) {
+  const action: TaskAction = {
+    type: actionTypes.UPDATE_TASK,
+    task,
+  };
+  return simulateHttpRequest(action);
+}
+
+export function removeTask(task: ITask) {
+  const action: TaskAction = {
+    type: actionTypes.REMOVE_TASK,
+    task,
+  };
+  return simulateHttpRequest(action);
+}
+
+export function simulateHttpRequest(action: TaskAction) {
```

Comment:
I think its better to put this at the top of the file since the other action creators are using this function